etcdserver/server.go

```diff
@@ -204,6 +204,13 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		cl *cluster
 	)
 
+	if mkerr := fileutil.MkdirAllIfNotExist(cfg.DataDir); mkerr != nil {
+		return nil, fmt.Errorf("cannot create data directory: %v", mkerr)
+	}
+	if wderr := fileutil.IsDirWriteable(cfg.DataDir); wderr != nil {
```

Comment:
If it does the `IsDirWriteable` test it'll change the mtime of the directory because of that ".touch" file business, so it all works out.